test/index.js

```diff
@@ -11,8 +11,8 @@ describe('when path is a directory', function () {
   it('should 404', function (done) {
     var app = koa()
     app.use(function* (next) {
-      var stats = yield* sendfile.call(this, __dirname)
-      assert.ok(stats)
+      sendfile(this, __dirname)
+      .then(assert.ok)
     })
```

Comment:
I'd rather just change `yield*` to `yield`, and remove `call`

If you don't `yield` sendfile, the execution order is nonlinear, even beyond the koa's life cycle.